include/revng/Model/ABI.h

```diff
@@ -235,28 +253,47 @@ inline constexpr model::ABI::Values getDefault(model::Architecture::Values V) {
   case model::Architecture::Invalid:
   case model::Architecture::Count:
   default:
-    revng_abort();
+    revng_abort("Unsupported architecture.");
   }
 }
 
-// TODO: Consider factoring these binary specific things into a ELF/PEModel.h.
 inline constexpr model::ABI::Values
-getDefaultMicrosoftABI(model::Architecture::Values V) {
+getDefaultForELF(model::Architecture::Values V) {
   switch (V) {
-  case model::Architecture::x86_64:
-    return model::ABI::Microsoft_x86_64;
   case model::Architecture::x86:
-    return model::ABI::Microsoft_x86_cdecl;
+    return model::ABI::SystemV_x86;
+  case model::Architecture::x86_64:
+    return model::ABI::SystemV_x86_64;
   case model::Architecture::mips:
     return model::ABI::SystemV_MIPS_o32;
   case model::Architecture::mipsel:
     return model::ABI::SystemV_MIPSEL_o32;
-  case model::Architecture::arm:
-    return model::ABI::AAPCS;
+  default:
+    return getDefaultFallback(V);
+  }
+}
+
+inline constexpr model::ABI::Values
+getDefaultForPECOFF(model::Architecture::Values V) {
+  switch (V) {
+  case model::Architecture::x86_64:
+    return model::ABI::Microsoft_x86_64;
+  case model::Architecture::x86:
+    return model::ABI::Microsoft_x86_cdecl;
   case model::Architecture::aarch64:
-    return model::ABI::AAPCS64;
+    return model::ABI::Microsoft_AAPCS64;
   default:
-    revng_abort();
+    return getDefaultFallback(V);
+  }
+}
+
+inline constexpr model::ABI::Values
+getDefaultForMachO(model::Architecture::Values V) {
+  switch (V) {
+  case model::Architecture::aarch64:
+    return model::ABI::Apple_AAPCS64;
+  default:
+    return getDefaultFallback(V);
   }
 }
```

Comment:
`getDefaultFallback` handles x86-64 ABIs, correct?

Reply:
`getDefaultFallback` is the "current default" in a way. This lets you only specify "overrides" for the architectures that needs something special to limit duplication.

Reply:
I guess we could be even stricter with it and explicitly disallow some combinations (for example MIPS in a Mach-O binary), but for now I opted to preserve the current behaviour.